ably/realtime_presence_test.go

```diff
@@ -96,3 +104,38 @@ func TestRealtimePresence_Sync250(t *testing.T) {
 		t.Fatalf("members3: %v", err)
 	}
 }
+
+var stateGuard = ^ably.StateEnum(0)
+
+var presTransitions = []ably.StateEnum{
+	ably.StateConnConnecting,
+	ably.StateConnConnected,
+	stateGuard,
+	ably.StateChanAttaching,
+	ably.StateChanAttached,
+}
+
+func TestRealtimePresence_EnsureChannelIsAttached(t *testing.T) {
+	rec := ably.NewStateRecorder()
+	opts := rec.Options()
+	opts.NoConnect = true
+	app, client := testutil.ProvisionRealtime(nil, opts)
+	defer safeclose(t, client, app)
+	channel := client.Channels.Get("persisted:presence_fixtures")
+	if err := ably.Wait(client.Connection.Connect()); err != nil {
+		t.Fatal(err)
+	}
+	rec.Add(stateGuard)
```

Comment:
Hm, sorry, I don't see it. I'll recap: I need to check in the test that StateChanAttached after the call to Presence.Get. Since the event sequence is [Connecting, Connected, Attaching, Attached] I need to somehow differentiate scenarios where I have attached chanel before `Presence.Get` or after `Presence.Get` (where the latter is only correct); by inserting mark after connecting I ensure that I receive either:

-  [Connecting, Connected, Attaching, Attached, mark] for the former
-  [Connecting, Connected, mark, Attaching, Attached] for the latter

I see two options: either I use mark to ensure ordering or I start listening for events after Connect and before `Presence.Get` (but I would need to drain the events first to ensure nothing is hanging asynchronously). I think the latter is more code that's why I went with first option.

If there's other way of implementing the test I don't see it.